swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/Store.swift

```diff
@@ -167,6 +172,15 @@ public final class Store: ObservableObject {
     }
   }
 
+  func toggleResource(resource: String, enabled: Bool) {
+    tunnelManager.toggleResource(resource: resource, enabled: enabled)
+    var newSettings = settings
+    newSettings.disabledResources = tunnelManager.disabledResources
+    Task {
+      try await save(newSettings)
```

Comment:
I think if we convert `save` to sync this doesn't need to be spawned in a new `Task` and we could do that because `save` spawns a task?

But I'm not too familiar with swift concurrency model, so I'm a bit afraid that'll break something

Reply:
I'm not sure how it works either 🤷‍♀️ The fact that you can access `self` inside a dispatcher queue is so strange

Reply:
Saving / loading from tunnelManager can sometimes take a little bit of time. If that gets blocked, this is the UI thread so all UI updates to the menubar would be stuck.

Reply:
But doesn't  `save` already spawn a task?

Reply:
Yeah looks like it. This may be redundant then, if you remove the double Task and it compiles and works then it's probably ok.

You just need to make sure writing to the tunnel manager doesn't block the UI thread.